packages/hebao_v1/contracts/version1.1/security/SecurityModule.sol

```diff
@@ -0,0 +1,171 @@
+abstract contract SecurityModule is MetaTxModule
+{
+    using SignedRequest for ControllerImpl;
+
+    // The minimal number of guardians for recovery and locking.
+    uint constant public MIN_ACTIVE_GUARDIANS = 2;
+
+    event WalletLock(
+        address indexed wallet,
+        uint            lock
+    );
+
+    constructor(
+        ControllerImpl _controller,
+        address        _trustedForwarder
+        )
+        public
+        MetaTxModule(_controller, _trustedForwarder) {}
+
+    modifier onlyFromWallet(address wallet)
+        override
+    {
+        address payable _logicalSender = logicalSender();
+        // We DO accept the wallet owner as the sender on behalf of the wallet!!!
+        require(
+            _logicalSender == wallet || _logicalSender == Wallet(wallet).owner(),
+             "NOT_FROM_WALLET_OR_OWNER"
+        );
+        controller.securityStore().touchLastActive(wallet);
+        _;
+    }
+
+    modifier onlyFromGuardian(address wallet)
+    {
+        require(
+            controller.securityStore().isGuardian(wallet, logicalSender()),
+            "NOT_FROM_GUARDIAN"
+        );
+        _;
+    }
+
+    modifier onlyWhenWalletLocked(address wallet)
+    {
+        require(isWalletLocked(wallet), "NOT_LOCKED");
+        _;
+    }
+
+    modifier onlyWhenWalletUnlocked(address wallet)
+    {
+        require(!isWalletLocked(wallet), "LOCKED");
+        _;
+    }
+
+    modifier onlyWalletGuardian(address wallet, address guardian)
+    {
+        require(controller.securityStore().isGuardian(wallet, guardian), "NOT_GUARDIAN");
+        _;
+    }
+
+    modifier notWalletGuardian(address wallet, address guardian)
+    {
+        require(!controller.securityStore().isGuardian(wallet, guardian), "IS_GUARDIAN");
+        _;
+    }
+
+    modifier onlyHaveEnoughGuardians(address wallet)
+    {
+        require(
+            controller.securityStore().numGuardians(wallet) >= MIN_ACTIVE_GUARDIANS,
+            "NO_ENOUGH_ACTIVE_GUARDIANS"
+        );
+        _;
+    }
+
+    // ----- internal methods -----
+
+    function quotaStore()
+        internal
+        view
+        returns (address)
+    {
+        return address(controller.quotaStore());
+    }
+
+    function lockWallet(address wallet)
+        internal
+    {
+        lockWallet(wallet, controller.defaultLockPeriod());
+    }
+
+    function lockWallet(address wallet, uint _lockPeriod)
+        internal
+        onlyWhenWalletUnlocked(wallet)
+    {
+        // cannot lock the wallet twice by different modules.
+        require(_lockPeriod > 0, "ZERO_VALUE");
+        uint lock = now + _lockPeriod;
+        controller.securityStore().setLock(wallet, lock);
+        emit WalletLock(wallet, lock);
+    }
+
+    function unlockWallet(address wallet, bool forceUnlock)
+        internal
+    {
+        (uint _lock, address _lockedBy) = controller.securityStore().getLock(wallet);
+        if (_lock > now) {
```

Comment:
Add require here or emit Event that do not need unlock, else relayer don't know what happened and need query from the contract

Reply:
Fixed.